File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys


def input():
    return sys.stdin.readline().rstrip()


n = int(input())
sequence = [int(input()) for _ in range(n)]
stack = []
operators = []
cnt = 1
possible = True
for num in sequence:
    while cnt <= num:
        stack.append(cnt)
        operators.append('+')
        cnt += 1
    if stack[-1] == num:
        stack.pop()
        operators.append('-')
    else:
        possible = False
        break
if possible:
    for op in operators:
        print(op)
else:
    print('NO')

File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include <iostream>
#include <stack>
#include <vector>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n;
	cin >> n;
	stack<int> s;
	vector<char> op;
	int cnt = 1;
	while (n--) {
		int x;
		cin >> x;
		while (cnt <= x) {
			s.push(cnt);
			op.push_back('+');
			cnt++;
		}
		if (s.top() == x) {
			s.pop();
			op.push_back('-');
		}
		else {
			cout << "NO";
			return 0;
		}
	}
	for (char& c : op) {
		cout << c << '\n';
	}
}